lib/model/query/forms.js

```diff
@@ -172,6 +172,14 @@ const del = (form) => ({ run, Assignments }) =>
   Promise.all([ run(markDeleted(form)), Assignments.revokeByActeeId(form.acteeId) ]);
 del.audit = (form) => (log) => log('form.delete', form);
 
+const restore = (form) => ({ run, Forms }) =>
+  Forms.getByProjectAndXmlFormId(form.projectId, form.xmlFormId)
+    .then((activeForm) => (activeForm.isDefined()
+      ? reject(Problem.user.uniquenessViolation({ fields: 'xmlFormId', values: form.xmlFormId }))
```

Comment:
DB should catch this error itself so catch it there and use problem/translate thing to make useful error